packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -423,18 +418,15 @@ async function validateAttestationNoSignatureCheck(
     validatorIndex = (attestationOrCache.attestation as SingleAttestation<ForkPostElectra>).attesterIndex;
     // [REJECT] The attester is a member of the committee -- i.e.
     // `attestation.attester_index in get_beacon_committee(state, attestation.data.slot, index)`.
-    // If `aggregationBitsElectra` exists, that means we have already cached it. No need to check again
-    if (aggregationBits === null) {
-      // Position of the validator in its committee
-      const committeeValidatorIndex = committeeValidatorIndices.indexOf(validatorIndex);
-      if (committeeValidatorIndex === -1) {
-        throw new AttestationError(GossipAction.REJECT, {
-          code: AttestationErrorCode.ATTESTER_NOT_IN_COMMITTEE,
-        });
-      }
-
-      aggregationBits = BitArray.fromSingleBit(committeeValidatorIndices.length, committeeValidatorIndex);
+    // Position of the validator in its committee
+    const committeeValidatorIndex = committeeValidatorIndices.indexOf(validatorIndex);
+    if (committeeValidatorIndex === -1) {
+      throw new AttestationError(GossipAction.REJECT, {
+        code: AttestationErrorCode.ATTESTER_NOT_IN_COMMITTEE,
+      });
     }
+
+    aggregationBits = BitArray.fromSingleBit(committeeValidatorIndices.length, committeeValidatorIndex);
```

Comment:
aggregationBits is backed by Uint8Array which is not cheap so we want to avoid if possible

we don't need to return `aggregationBits` for electra, just return `committeeValidatorIndex`

at `attestationPool`, we only create `aggregationBits` once for the first attestation
attestations come after that just use `aggregate.aggregationBits.set(bitIndex, true);`

this can be applied for pre-electra too. So the interface of AttestationPool would be

```typescript
function aggregateAttestationInto(
  aggregate: AggregateFast,
  attestation: SingleAttestation,
  bitIndexInCommittee: number
): InsertOutcome {
```